staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -56,6 +61,11 @@ type proxyHandler struct {
 }
 
 func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	if r.isLocal() {
```

Comment:
> calling isLocal() and getDestinationHost() separately is racy if updateAPIService() is called in between... isLocal() could return false here, and getDestinationHost() could be invalid by the time you check it below

Sounds a lot like have your cake and eat it too.  The atomic bool would never be synced to anything below it.  Given the choice, I'd probably let a few calls fail during a transition.